### PR TITLE
Removing python for vsl.plot - WIP

### PR DESCRIPTION
**Pull request by ulises-jeremias**
- Updated vsl.plot to avoid the need of using python
- Updated vsl.plot to use plotly.js and vweb instead of plotly.py
- Fixed some CI checks
- Updated plot examples

### Summary by CodeRabbit

**New Features:**
- Introduced a more intuitive way to add different types of data visualizations (scatter, pie, heatmap, surface, bar, histogram) to a plot.
- Added a new `App` struct to generate an HTML page with a Plotly graph.
- Introduced a `port` constant to specify the server port.

**Improvements:**
- Renamed the `plot` module to `plotly` for clarity.
- Replaced `vsl.errors` import with `vweb` for better error handling.
- Renamed several functions (e.g., `add_trace` to `scatter3d`, `add_annotation` to `annotation`, `set_layout` to `layout`) for consistency and clarity.
- Updated examples and tests to reflect the new function names and usage.

**Refactor:**
- Removed `create_venv` and `solve_mod_path` functions.
- Updated `Trace` to represent different trace types.
- Modified `show` function to use `vweb.run` to start the server.

These changes enhance the software's intuitiveness and flexibility, making it easier for users to create and customize their data visualizations.

### CODE REVIEW

File: examples/plot_saddle_surface/main.v [hunk 1]
```diff
@@ -18,14 +18,13 @@ fn main() {
...
 		x: x
 		y: y
 		z: z
 		colorscale: 'Viridis'
 	)
-	plt.set_layout(
+	plt.layout(
 		title: 'Saddle Surface'
 	)
 	plt.show()!
```
[the remark this answers is not in the document]
Comment (ulises-jeremias) [reply examples/plot_saddle_surface/main.v:30]: @coderabbitai this is not Golang!

File: plot/show.v [hunk 2]
```diff
@@ -1,70 +1,82 @@
...
-	}
-	init_path := solve_mod_path('scripts', 'create-venv.sh')
-	result := os.execute('bash ${init_path} "${venv_path}"')
-	if result.exit_code != 0 {
-		return errors.error(result.output, .efailed)
-	}
+fn preprocess_data_string(data string) string {
+	return data.replace(',"[]f64"', '').replace('"[]f64"', '').replace(',"[]string"',
+		'""').replace('"[]string"', '""')
 }
```
[the remark this answers is not in the document]
Comment (JalonSolov) [reply plot/show.v]: @coderabbitai V does not support regex in string `replace`.  Only in regex `replace`.

**Comment by ulises-jeremias** [comment]
@coderabbitai pause

**Comment by ulises-jeremias** [comment]
@coderabbitai pause

File: plot/show.v [hunk 3]
```diff
@@ -1,70 +1,93 @@
...
+	traces_with_type := app.plot.traces.map({
+		'type':  TracesWithTypeValue(it.trace_type())
+		'trace': TracesWithTypeValue(it)
+	})
+	traces_with_type_json := encode(traces_with_type)
+	layout_json := encode(app.plot.layout)
+	go fn () {
+		time.sleep(100 * time.millisecond)
+		exit(0)
+	}()
```
[the remark this answers is not in the document]
Comment (ulises-jeremias) [reply plot/show.v:52]: https://github.com/coderabbitai pause

[the remark this answers is not in the document]
Comment (ulises-jeremias) [reply plot/show.v:52]: @coderabbitai pause

**Comment by ulises-jeremias** [comment]
https://github.com/vlang/vsl/issues/115 related

**Pull request merged**